src/games/strategy/triplea/ai/fastAI/FastAI.java

```diff
@@ -9,14 +9,16 @@
  */
 public class FastAI extends ProAI {
 
+  // Odds estimator
+  private final static IOddsCalculator estimator = new FastOddsEstimator();
+
   public FastAI(final String name, final String type) {
     super(name, type);
   }
 
   @Override
-  public void initializeBattleCalculator() {
-    final IOddsCalculator calc = new FastOddsEstimator();
-    ProBattleUtils.setOddsCalculator(calc);
+  protected void initializeCalc() {
+    calc = new ProOddsCalculator(estimator);
```

Comment:
You could move that code to a factory (logically fits better there) but its only used in this one place so not a huge deal. I think the difference has been that traditionally each AI implementation has essentially been completely separate so weak AI, strong AI, dynamix AI, and pro AI all had separate code bases. They were written by different developers essentially in isolation.

Well you could either extend the existing AIs or create a completely separate one. If you extend ProAI then you could use the concurrent calc and if you extend FastAI then you could use the new estimator. So yeah you'd have an inheritance structure which for AIs I think is fine.